src/commands/dev/exec.js

```diff
@@ -1,42 +1,22 @@
-const fromEntries = require('@ungap/from-entries')
-const chalk = require('chalk')
 const execa = require('execa')
 
 const Command = require('../../utils/command')
-const { addEnvVariables, getEnvSettings } = require('../../utils/env')
-const {
-  // NETLIFYDEV,
-  NETLIFYDEVLOG,
-  // NETLIFYDEVWARN,
-  NETLIFYDEVERR,
-} = require('../../utils/logo')
+const { getSiteInformation, addEnvVariables } = require('../../utils/dev')
 
 class ExecCommand extends Command {
   async run() {
-    const { site, api } = this.netlify
-    if (site.id) {
-      // just to show some visual response first
-      this.log(`${NETLIFYDEVLOG} Checking your site's environment variables...`)
-      await addEnvVariables(api, site)
-    } else {
-      this.log(
-        `${NETLIFYDEVERR} No Site ID detected. You probably forgot to run \`netlify link\` or \`netlify init\`. `,
-      )
-    }
-
-    const envSettings = await getEnvSettings({ projectDir: site.root, warn: this.warn })
-    if (envSettings.vars.length !== 0) {
-      this.log(
-        `${NETLIFYDEVLOG} Adding the following env variables from ${envSettings.files.map((file) =>
-          chalk.blue(file),
-        )}:`,
-        chalk.yellow(envSettings.vars.map(([key]) => key)),
-      )
-    }
+    const { log, warn, error, netlify } = this
+    const { site, api } = netlify
+    const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
+      api,
+      site,
+      warn,
+      error,
+    })
+    await addEnvVariables({ log, teamEnv, addonsEnv, siteEnv, dotFilesEnv })
 
     execa(this.argv[0], this.argv.slice(1), {
       stdio: 'inherit',
-      env: fromEntries(envSettings.vars),
```

Comment:
These are applied globally now (which is not nice, but would require more refactoring as we log each env that is applied in a different way)

Reply:
Would it be possible to keep those environment variables into a variable, instead of modifying `process.env`?

Reply:
`addEnvVariables` modifies `process.env` but also prints log messages while doing so:
https://github.com/netlify/cli/pull/1490/files#diff-fc861c54030ca23b6d43638aa92f3f446bca1b187ce3b3fea5b8e6d7d1c0a19cR98

We can clone `process.env` in `addEnvVariables`, use the clone instead of `process.env` and return it to `dev:exec`.
That would work for `dev:exec` but we would still need to modify `process.env` in other places due to the way `netlify dev` and `netlify functions:create` command work at the moment.

I would rather be consistent in how we're applying env variables and handle mutating `process.env` (in all locations we're doing it) in another PR. WDYT?

Reply:
This makes sense :+1: 
Should we create an issue for it?

Reply:
https://github.com/netlify/cli/issues/1517